sdk-commons/src/main/java/org/openmrs/maven/plugins/utility/DistributionBuilder.java

```diff
@@ -137,7 +144,14 @@ protected void populateRefApp3xProperties(Distribution distribution, Properties
 		// Add spa properties if they are not included explicitly
 		if (includedProperties.getSpaProperties().isEmpty()) {
 			Properties frontendProperties;
-			if (new Version(distroVersion).higher(new Version("3.0.0-beta.16"))) {
+			if (spaConfigFile != null) {
+                try {
+                    frontendProperties = PropertiesUtils.getFrontendPropertiesFromJson(Files.newInputStream(Paths.get(spaConfigFile)));
+                } catch (IOException e) {
+                    throw new RuntimeException(e);
+                }
+            }
+			else if (new Version(distroVersion).higher(new Version("3.0.0-beta.16"))) {
```

Comment:
The fate of this will depend on the other decisions we make in the questions I raise above, but we should review the hacks that already exist to support refapp 3.x.x here as we look at how best to support these use cases moving forward.